lib/matplotlib/tests/test_collections.py

```diff
@@ -78,7 +78,8 @@ def test__EventCollection__get_props():
     # check that the default lineoffset matches the input lineoffset
     assert props['lineoffset'] == coll.get_lineoffset()
     # check that the default linestyle matches the input linestyle
-    assert coll.get_linestyle() == [(0, None)]
+    assert coll.get_linestyle() == ['-']
```

Comment:
Changing the return value of `get_linestyle()` is almost certainly causing trouble.

I think we at least need a forward note that this will change, plus maybe helpers `to_linestyle()`, so that users can do
```
to_linestyle(coll.get_linestyle()) == ['-']
```
to be compatible with all versions of matplotlib.

Reply:
Currently, the return type of `get_linestyle` is inconsistent between classes. Collections return patterns, Patches returns "solid" etc, Lines returns "-" etc... 

Collections have an alias for dashes, which is equal to linestyle, but the others do not have a `get_dashes` method.

Reply:
I agree that we should consolidate this, but changing return values is a breaking API change, which we need to do as carefully as possible.